Document Java Backpressure Option (#9539)
getsentry/sentry-docs: docs/platforms/java/common/configuration/options.mdx

</ConfigKey>

<ConfigKey name="enable-backpressure-handling">

> Set this boolean to `false` to disable automatic downsampling of transactions while the system is under load. The `tracesSampleRate` is halved for every failing health check up to 10 times, roughly meaning 0.001% of the original `tracesSampleRate`. Any positive health check will reset `tracesSampleRate` to its original value. Health checks are happening in the background every 10 seconds and check for queue drops as well as rate limiting.

Reviewer: ```suggestion
Set this boolean to `false` to disable automatic downsampling of transactions while the system is under load. The `tracesSampleRate` is halved for every failing health check up to 10 times, (roughly 0.001% of the original `tracesSampleRate`). Any positive health check will reset `tracesSampleRate` to its original value. Health checks run in the background every 10 seconds, checking for queue drops and rate limiting.
```

> _(New in version 7.1.0)_

Author: Maybe it's sufficient to add another sentence mentioning this used to be opt-in in 7.1.

> _(New in version 7.1.0)_

Reviewer 2: Ah, now I get it. It was introduced in `7.1.0` and with `7.8.0` the default has changed from `false` to `true`.

> Set this boolean to `false` to disable automatic downsampling of transactions while the system is under load. The `tracesSampleRate` is halved for every failing health check up to 10 times, (roughly 0.001% of the original `tracesSampleRate`). Any positive health check will reset `tracesSampleRate` to its original value. Health checks run in the background every 10 seconds, checking for queue drops and rate limiting. Note: Before version 7.8.0 backpressure handling was disabled by default.

Reviewer 2: (l) Hmm, does that make it easier to understand? I was initially confused about the note. But am not sure if that is much better tbh.

```suggestion
Set this boolean to `false` to disable automatic downsampling of transactions while the system is under load. The `tracesSampleRate` is halved for every failing health check up to 10 times, (roughly 0.001% of the original `tracesSampleRate`). Any positive health check will reset `tracesSampleRate` to its original value. Health checks run in the background every 10 seconds, checking for queue drops and rate limiting. Note: Starting with version 7.8.0 backpressure handling has changed from opt-in to opt-out.
```
